unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/ChatWidgetHUD/ChatProfanityFeatureFlagFromDataStore.cs

```diff
@@ -0,0 +1,6 @@
+﻿using DCL;
+
+public class ChatProfanityFeatureFlagFromDataStore : IChatProfanityFeatureFlag
```

Comment:
I agree its a bit overhead. The main idea was to avoid referencing a singleton inside the controller but also solves the feature deactivation from `PrivateChatWindowHUDController`. I could remove `IChatProfanityFeatureFlag` and its implementations like this `public ChatHUDController(RegexProfanityFilter profanityFilter = null)` and then checking for null on usage.
What do you think of injecting `DataStore` into `ChatHUDController` constructor instead? I could manipulate the data instance in the test, and also pass `DataStore.i` on production code

Reply:
I like the idea, it also aligns really well with the future Plugin System. @BrianAmadori could we have your input here?

Reply:
I just uploaded these changes, so you can check it in the code